### PR TITLE
[16.0][ADD] delivery_driver

### PR DESCRIPTION
**Pull request by EmilioPascual**
Allow you to assign default driver in delivery carriers and pickings.

https://www.loom.com/share/78a5e47d4ed2453ab469df3217a5b83b

@Shide, @rafaelbn, @Gelojr please review, if you want.

@moduon MT-4427

### CODE REVIEW

**Review by fcvalgar — APPROVED** [review]
Functional review,

LGTM, than you @EmilioPascual 


image


image

image

**Review by Gelojr — APPROVED** [review]
LGTM In addition to @EmilioPascual tests, I have checked the following:
- If a driver is added to a shipping method where the field has no value, the delivery notes already created are not updated.
- You can change the driver in bulk from the transfers list view, but not from the shipping methods list view.
- Quick search and grouping by driver is available from the shipping methods and transfers list views.
https://www.loom.com/share/56307e5e065d400fb949c06bff1bb3fb?sid=a7b32eb8-9b98-4a0c-b68c-0afae7150966

File: delivery_driver/views/delivery_carrier.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 Moduon Team S.L.
+     License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0) -->
+<data>
```
Comment (rousseldenis) [inline delivery_driver/views/delivery_carrier.xml]: Please add `<odoo>` as root

File: delivery_driver/models/stock_move_line.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo import fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    driver_id = fields.Many2one(related="picking_id.driver_id", store="True")
```
Comment (rousseldenis) [inline delivery_driver/models/stock_move_line.py]: Should it really be stored ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]

File: delivery_driver/models/stock_move_line.py (hunk 2 above)
Comment (rafaelbn) [reply delivery_driver/models/stock_move_line.py]: Is needed to group by

**Review by FrancoMaxime — COMMENTED** [review]
LGTM: code review

File: delivery_driver/views/delivery_carrier.xml (hunk 1 above)
Comment (EmilioPascual) [reply delivery_driver/views/delivery_carrier.xml]: I don't understand you. What do you mean?

File: delivery_driver/models/stock_move_line.py (hunk 2 above)
Comment (EmilioPascual) [reply delivery_driver/models/stock_move_line.py]: Store is necessary to group by driver

**Comment by rafaelbn** [comment]
Hello @rousseldenis ! Happy new year! 😄 

@EmilioPascual answered you, could you please check it to solve doubts?

Thank you! ❤️

File: delivery_driver/views/delivery_carrier.xml (hunk 1 above)
Comment (rousseldenis) [reply delivery_driver/views/delivery_carrier.xml]: Github obfuscated my comment 😅

Comment (EmilioPascual) [reply delivery_driver/views/delivery_carrier.xml]: Ok, no problem. It's changed. Thank you.

File: delivery_driver/models/stock_move_line.py (hunk 2 above)
Comment (Shide) [inline delivery_driver/models/stock_move_line.py]: If someone wants to choose another delivery driver, domain must be present.

```suggestion
    driver_id = fields.Many2one(related="picking_id.driver_id", store="True", domain="[('is_company', '=', False)]")
```

**Review by Shide — CHANGES_REQUESTED** [review]
Good job!

**suggestion**: Add domain on stock.move.line too

File: delivery_driver/models/stock_move_line.py (hunk 2 above)
Comment (EmilioPascual) [reply delivery_driver/models/stock_move_line.py]: 👌🏻 Added

File: delivery_driver/views/delivery_carrier.xml (hunk 1 above)
Comment (EmilioPascual) [reply delivery_driver/views/delivery_carrier.xml]: @rousseldenis can you review again, please?
Thanks 😊

**Review by Shide — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]
👍🏼 Let's go!

**Comment by rafaelbn** [comment]
@rousseldenis this PR mark " rousseldenis requested changes " but you have already solved question. Is it Ok for you if I merge? Thank you! 😄

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

Solved

imagen

**Comment by rousseldenis** [comment]
> @rousseldenis this PR mark " rousseldenis requested changes " but you have already solved question. Is it Ok for you if I merge? Thank you! 😄

Sorry, was ill last week :cold_sweat:

**Comment by EmilioPascual** [comment]
> > @rousseldenis this PR mark " rousseldenis requested changes " but you have already solved question. Is it Ok for you if I merge? Thank you! 😄
> 
> Sorry, was ill last week 😰

Thanks @rousseldenis, I hope you are feeling better.

**Pull request merged**